.github/workflows/snarkyjs-api-reference.yml

```diff
@@ -0,0 +1,56 @@
+name: SnarkyJS-API-Reference
+on:
+  workflow_dispatch:
+    snarkyjs_branch:
+      type: choice
+      description: SnarkyJS branch to build from
+      options:
+        - main
+        - releases
+jobs:
+  update-snarkyjs-api-reference:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Checkout Docs repo
+        uses: actions/checkout@v3
+        with:
+          path: docs
+
+      - name: Checkout SnarkyJS repo
+        uses: actions/checkout@v3
+        with:
+          repository: o1-labs/snarkyjs
+          path: snarkyjs
+          ref: ${{ github.event.inputs.${{ github.event.inputs.message }} }}
+
+      - name: Build API reference docs
+        run: |
+          # Install dependencies and build SnarkyJS
+          cd snarkyjs && npm ci && npm run build
+
+          # Install markdown-plugin for typedoc and build the SnarkyJS API reference
+          npm install typedoc-plugin-markdown
+          npx typedoc --readme none --name SnarkyJS --plugin typedoc-plugin-markdown --out snarkyjs-reference src/index.ts --hideBreadcrumbs true
+          rm snarkyjs-reference/.nojekyll
+
+          # Install dependencies for Docs repo
+          cd ../docs/ && npm ci
+
+          # Remove the old API reference and copy the newly generated reference
+          rm -rf docs/zkapps/snarkyjs-reference && cp -rv ../snarkyjs/snarkyjs-reference docs/zkapps/snarkyjs-reference
+
+          # Update with new enteries and format sidebars.js
+          node update-sidebars-snarkyjs-api.js && npx prettier --config .prettierrc --write docusaurus.config.js
+
+          # Configure git and push
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add docs/zkapps/snarkyjs-reference
+          git add docusaurus.config.js
+          git commit -m "Build and publish new SnarkyJS API reference docs"
+          git push
```

Comment:
Any benefit to PR'ing it in stead of pushing directly? This current way is more similar to how it will need to be set up for the eventually fully automated solution, assuming it works with branch protection rules. Lmk if we need to tweak those.

Reply:
That's a good point! I suppose the benefit of creating a PR would be that you can create a deploy link and inspect changes if you wish. I will change it to create a PR instead :D